src/main/java/com/powsybl/openloadflow/ac/AbstractTransformerVoltageControlOuterLoop.java

```diff
@@ -48,4 +50,43 @@ public void cleanup(LfNetwork network) {
             branch.getVoltageControl().ifPresent(voltageControl -> branch.setVoltageControlEnabled(true));
         }
     }
+
+    protected void checkControl(LfNetwork network) {
+        List<LfBranch> controllerBranches = new ArrayList<>(1);
+        List<LfBranch> disabledBranches = new ArrayList<>(1);
+        for (LfBranch branch : network.getBranches()) {
+            if (!branch.isDisabled() && branch.isVoltageController() && branch.isVoltageControlEnabled()) {
+                controllerBranches.add(branch);
+            }
+            if (branch.isDisabled()) {
+                disabledBranches.add(branch);
+            }
+        }
+        for (LfBranch branch : controllerBranches) {
+            var connectivity = network.getConnectivity();
+            var voltageControl = branch.getVoltageControl().orElseThrow();
+            var controlledBus = voltageControl.getControlled();
+            // apply contingency (in case we are inside a security analysis)
+            for (LfBranch disabledBranch : disabledBranches) {
+                connectivity.cut(disabledBranch.getBus1(), disabledBranch.getBus2());
+            }
+            connectivity.cut(branch.getBus1(), branch.getBus2());
+            Set<LfBus> componentOnNotControlledSide = null;
+            if (controlledBus.equals(branch.getBus1())) {
+                componentOnNotControlledSide = connectivity.getConnectedComponent(branch.getBus2());
+            } else if (controlledBus.equals(branch.getBus2())) {
+                componentOnNotControlledSide = connectivity.getConnectedComponent(branch.getBus1());
+            } else {
+                // I don't know.
+            }
+            if (componentOnNotControlledSide != null) {
+                Optional<LfBus> generatorControlledBus = componentOnNotControlledSide.stream().filter(LfBus::isVoltageControlled).findAny();
+                if (!generatorControlledBus.isPresent()) {
+                    branch.setVoltageControlEnabled(false);
```

Comment:
Do not forget to restore it in the outer loop cleanup!

Reply:
It is done as the condition is only to have a voltage control, no?